File: frontend/src/models/cat.ts
import { Breed } from './breed';
import { Sex } from './sex';
import { Client } from './client';
import { Food } from './food';
import { Allergen } from './allergen';


export interface Cat {
    id?: number;
    name: string;
    breed?: Breed;
    birthday?: Date;
    sex: Sex;
    color: number;
    owner?: Client;
    preferences: Food[];
    allergens: Allergen[];
}

File: src/main/kotlin/ru/byprogminer/dbcw/entity/Cat.kt
package ru.byprogminer.dbcw.entity

import java.time.LocalDateTime
import javax.persistence.*

@Entity
class Cat(
    @Id @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "cat_id_seq")
    @SequenceGenerator(name = "cat_id_seq", sequenceName = "cat_id_seq", allocationSize = 1)
    val id: Long?,

    var name: String,

    @ManyToOne
    var breed: CatBreed?,

    var birthday: LocalDateTime?,

    @Enumerated(EnumType.STRING)
    var sex: Sex,

    var color: Int,

    @ManyToOne
    var owner: Client?
) {

    constructor(): this(null, "", null, null, Sex.M, 0, null)

    enum class Sex {
        M, F
    }
}
